homeassistant/components/pi_hole/config_flow.py

```diff
@@ -34,19 +37,21 @@ class PiHoleFlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
 
     VERSION = 1
 
-    def __init__(self):
+    def __init__(self) -> None:
         """Initialize the config flow."""
-        self._config = None
+        self._config: dict = {}
 
-    async def async_step_user(self, user_input=None):
+    async def async_step_user(self, user_input: dict | None = None) -> FlowResult:
```

Comment:
We normally type the input like this:
```suggestion
    async def async_step_user(
        self, user_input: dict[str, Any] | None = None
    ) -> FlowResult:
```